### DISCUSSION TITLE
Time Zone backwards

### DISCUSSION THREAD

**Discussion by chirpp2** [netbox-community/netbox#18168] [Help Wanted!]
Hello, i had been planning to use the settings for Time Zone when adding a site but I need it to be UTC-# (UTC-8, for example.) No options exists and etc/gmt-8 is what I thought would it should be but that is reversed! All of our network gear have the UTC option so to use etc/gmt i would need to enter etc/gmt+8 to get the correct time zone for a device in utc-8.  We also wanted to pull this field into other systems and be able to automate according to th time zone, or confirm that setting is accurate on devices, but with the mismatch/reverse that is messy.  
How can i get UTC time zones into the field? It appears they pull from zoneinfo web site so I am not sure i can modify this.  is it an option to hide this field and create a enw field with choices of time zones ? That would work but then in the summary page for the site, it does not show the current time which is handy when dealing with a site.

**Comment by candlerb** [comment]
What actual country/area are you in? Choose the timezone appropriate for that country. Then you'll also get any daylight savings adjustments if required.

The POSIX `Etc` timezones are indeed weird and should be avoided. They are described in Wikipedia [here](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones):

> "Sign is intentionally inverted. See the [Etc area description](https://en.wikipedia.org/wiki/Tz_database#Area)"

In general, the best approach for interoperability between systems is to use UTC everywhere. Then you use the user's local timezone for *display* of times only, i.e. converting the stored UTC in database to local time in web pages.

**Comment by chirpp2** [comment]
Thanks for the reply!  
I should be more specific. This is in the Sites area that I am filling in the time zone.  As an example we have a site in California so is UTC-8.  A site in France so UTC+1.  The idea is I can take those numbers with some automation and make decisions on where to send syslogs to, for example. I think the Time zone you are referring to is within netbox settings for the gui and i have that set already.  
I would choose America/LosAngeles and Europe/Paris but then this leaves this inofmration subjective when the city is not a well know city, what area is it nearby, etc. Make sense?

**Reply by candlerb** [reply]
> I would choose America/LosAngeles and Europe/Paris but then this leaves this inofmration subjective when the city is not a well know city, what area is it nearby, etc

That's exactly what you should do. For each locality, there may be only one significant city, or a few. In the latter case, pick any one. It's not making an assertion about the physical location, it's just saying "uses the same timezone as this city".

Unix timezones work like this (using cities rather than countries or states) because for historical reasons, certain cities have moved between timezones - and prior to timezone standardization had their own local solar time. It probably doesn't matter unless you're entering dates and times from the 19th Century, but tzdata supports it anyway, so that's how it works.

If you don't like this, you can ignore it, and/or add your own custom field on Site.

**Reply by chirpp2** [reply]
Understood. I was hoping to have it more definitive, but understand.  Daylight Savings is a concern if we are using tis field for actual time, and that is a concern when using UTC or GMT. 

I will confirm how we want to use this field and either create a custom field with choices or keep this as it is. 

I really appreciate your feedback!

**Reply by tyler-8** [reply]
The timezone names are directly from the de-facto standard timezone database. You choose the one that matches the precise location of your sites.

https://en.wikipedia.org/wiki/Tz_database
https://en.wikipedia.org/wiki/List_of_tz_database_time_zones

You can use https://pypi.org/project/timezonefinder/ to find the exact match for a given site.
